src/main/Extensions/DataProtection.cs

```diff
@@ -70,4 +70,35 @@ public static class DataProtectionExtensions
             return Encoding.UTF8.GetString(clearBytes);
         }
     }
+    /// <summary>
+    /// forces a re-calculation of the protected data according to current machine setting in EncryptConfig when
+    /// writing the json for renewals and options for plugins
+    /// </summary>
+    public class protectedStringConverter : JsonConverter<string>
+    {
+        public override void WriteJson(JsonWriter writer, string protectedStr, JsonSerializer serializer)
+        {
+            try
+            {
+                string unprotected = protectedStr.Unprotect();
+                writer.WriteValue(unprotected.Protect());
+            }
+            catch
+            {
+                //couldn't unprotect string; keeping old value
+                writer.WriteValue(protectedStr);
+                writer.WriteComment("This protected string cannot be decrypted on current machine. See instructions about migrating to new machine.");
+            }
+        }
+        public override string ReadJson(JsonReader reader, Type objectType, string existingValue, bool hasExistingValue, JsonSerializer serializer)
+        {
+            const string clearPrefix = "clear-";
```

Comment:
I don't see the purpose of this prefix - it's not used by any other code?

Reply:
I put this in for the following usage scenario: a user moves all the files over to a new machine and no longer has access to the old machine. They realize--too late--that their passwords are encrypted in a machine-dependent format. This would allow the user to edit the json file manually and enter a clear-text password so that they can continue to use the renewal file. The alternative is that the user could enter the password as base64 (the way we store un-encrypted passwords) and achieve the same result.